nengo/utils/lock.py

```diff
@@ -0,0 +1,51 @@
+import errno
+import os
+import os.path
+import time
+
+
+class TimeoutError(Exception):
```

Comment:
For simplicity I would think all exceptions in Nengo should be a subclass of `NengoException`, rather than having to figure out when it should be and when it shouldn't